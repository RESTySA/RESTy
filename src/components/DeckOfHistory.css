.DeckOfHistory {
    width:300px;
    padding: 0 10px;

    h3 {
        margin:8px 0;
    }

    ul {
        margin: 10px 0;
        padding:0;
    }

    li {
        list-style: none;
        background-color:#2d5986e7;
        border: 1px solid black;
        border-radius: 2.5%;
        color: white;
        padding:4px;
        margin-bottom: 8px;


        h4{
            font-size:.8em;
            margin:0px;
        }
        span{
            font-size: .8em;
        }

    }

    li:hover{
        background-color: lightgrey;
        color:black;
    }
}