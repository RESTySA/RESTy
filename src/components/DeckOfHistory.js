import React from 'react';
import PropTypes from 'prop-types';
import History from './History';

const DeckOfHistory = (history) => {
  // console.log(history.history);
  let historyCard;
  if(history.length > 0) {
    console.log(history);
    historyCard = history.map(card => {
      console.log(card);
      return (
        <li key={card.id}>
          <History url={card.url}  method={card.method}/>
        </li>
      );
    });
  }

  return (
    <ul>
      {historyCard}
    </ul>
  );
};

DeckOfHistory.propTypes = {
  history: PropTypes.arrayOf(PropTypes.object)
};

export default DeckOfHistory;