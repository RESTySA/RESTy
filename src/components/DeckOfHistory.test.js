import React from 'react';
import { shallow } from 'enzyme';
import DeckOfHistory from './DeckOfHistory';

describe('DeckOfHistory component', () => {
  const test = [{
    id: '5da237699734fdcb7bef8f51',
    url: 'test.com',
    method: 'get'
  },
  {
    id: '5da237699734fdcb7bef8f52',
    url: 'test.com',
    method: 'post'
  }
  ];

  it('renders DeckOfHistory', () => {
    const wrapper = shallow(<DeckOfHistory history={test} />);
    expect(wrapper).toMatchSnapshot();
  });
});