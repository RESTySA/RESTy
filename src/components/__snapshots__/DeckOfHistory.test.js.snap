// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckOfHistory component renders DeckOfHistory 1`] = `
<div
  className="DeckOfHistory"
>
  <h3>
    History
  </h3>
  <ul>
    <li
      key="5da237699734fdcb7bef8f51"
    >
      <History
        method="get"
        url="test.com"
      />
    </li>
    <li
      key="5da237699734fdcb7bef8f52"
    >
      <History
        method="post"
        url="test.com"
      />
    </li>
  </ul>
</div>
`;
